Lib/subprocess.py

```diff
@@ -491,7 +491,13 @@ def run(*popenargs,
             stdout, stderr = process.communicate(input, timeout=timeout)
         except TimeoutExpired:
             process.kill()
```

Comment:
Why don't we close all pipes here? Closing pipes + .wait() should be enough, no? SIGKILL cannot be ignored on Unix (I don't know for Windows).

Reply:
Agreed this implementation is too complex.  That sounds tempting and simple.  Trying that, I found that it caused the capture_output_timeout tests to fail as we no longer collected the output and attached it to the Timeout exception.  But that led to some code inspection... we're already gathering the output within the original _communicate() on POSIX and can attach everything we have in the _original_ Timeout exception so that we don't need to call communicate() again after kill() at all in this path.

Windows collects output in a different manner so it still needs it, but this bug is POSIX only so I've reworked this to simplify it.  The communicate call for output gathering becomes Windows only.  I'm relying on the CI systems to do my Windows testing.